train.py

```diff
@@ -471,10 +471,16 @@ def train(hyp,  # path/to/hyp.yaml or hyp dictionary
 
             # Update best mAP
             fi = fitness(np.array(results).reshape(1, -1))  # weighted combination of [P, R, mAP@.5, mAP@.5-.95]
-            if fi > best_fitness or sparseml_wrapper.reset_best(epoch):
-                best_fitness = fi
+            save_names = sparseml_wrapper.get_best_save_names(epoch)
+            for name in save_names:
+                if name not in best_fitness:
+                    best_fitness[name] = -1
+
+                if fi >= best_fitness[name]:
+                    best_fitness[name] = fi
```

Comment:
I have a question, won't this part update the fitness for all names in save_names with the best_overall fitness? As in where are we checking if we are within pruning epochs or quantization_epochs?